src/replica/HttpAsyncClientApp.cc

```diff
@@ -0,0 +1,204 @@
+
+namespace lsst {
+namespace qserv {
+namespace replica {
+
+HttpAsyncClientApp::Ptr HttpAsyncClientApp::create(int argc, char* argv[]) {
+    return Ptr(new HttpAsyncClientApp(argc, argv));
+}
+
+
+HttpAsyncClientApp::HttpAsyncClientApp(int argc, char* argv[])
+    :   Application(
+            argc, argv,
+            ::description,
+            ::injectDatabaseOptions,
+            ::boostProtobufVersionCheck,
+            ::enableServiceProvider
+        ) {
+
+    parser().required(
+        "url",
+        "The URL to read data from.",
+        _url
+    ).option(
+        "method",
+        "The HTTP method. Allowed values: GET, POST, PUT, DELETE.",
+        _method
+    ).option(
+        "header",
+        "The HTTP header to be sent with a request. Note this test application allows"
+        " only one header. The format of the header is '<key>[:<val>]'.",
+        _header
+    ).option(
+        "data",
+        "The data to be sent in the body of a request.",
+        _data
+    ).option(
+        "max-response-data-size",
+        "The maximum size (bytes) of the response body. If a value of the parameter is set"
+        " to 0 then the default limit of 8M imposed by the Boost.Beast library will be assumed.",
+        _maxResponseBodySize
+    ).option(
+        "expiration-ival-sec",
+        "A timeout to wait before the completion of a request. The expiration timeout includes"
+        " all phases of the request's execution, including establishing a connection"
+        " to the server, sending the request and waiting for the server's response."
+        " If a value of the parameter is set to 0 then no expiration timeout will be"
+        " assumed for the request.",
+        _expirationIvalSec
+    ).option(
+        "file",
+        "A path to an output file where the response body received from a remote source will"
+        "  be written. This option is ignored if the flag --body is not specified.",
+        _file
+    ).flag(
+        "result2json",
+        "If specified the flag will cause the application to interpret the response body as"
+        " a JSON object.",
+        _result2json
+    ).flag(
+        "verbose",
+        "The flag that allows printing the completion status and the response header"
+        " info onto the standard output stream.",
+        _verbose
+    ).flag(
+        "body",
+        "The flag that allows printing the complete response body. If the --file=<path> option"
+        " is specified then the body will be written into that files. Otherwise it will be"
+        " printed onto the standard output stream.",
+        _body
+    );
+}
+
+
+int HttpAsyncClientApp::runImpl() {
+    string const context = "HttpAsyncClientApp::" + string(__func__) + "  ";
+    const char* allowedMethods[] = {"GET", "POST", "PUT", "DELETE"};
+    if (count(cbegin(allowedMethods), cend(allowedMethods), _method) == 0) {
+        throw invalid_argument(context + "unknown HTTP method: " + _method);
+    }
```

Comment:
Command line parser should do this check.

Reply:
You're 100% right. The command-line parser supports this. I will fix this.